src/Stratis.Bitcoin/Consensus/ConsensusManger.cs

```diff
@@ -74,13 +117,388 @@ public class ConsensusManager
+                    onBlockDownloadedCallback(null);
+                    continue;
+                }
+
+                if (chainedHeaderBlock.Block != null)
+                {
+                    this.logger.LogTrace("Block pair '{0}' was found in memory.", chainedHeaderBlock);
+                    onBlockDownloadedCallback(chainedHeaderBlock);
+                    continue;
+                }
+
+                if (this.blockStore != null)
+                {
+                    Block block = await this.blockStore.GetBlockAsync(blockHash);
+                    if (block != null)
+                    {
+                        var newBlockPair = new ChainedHeaderBlock(block, chainedHeaderBlock.ChainedHeader);
+                        this.logger.LogTrace("Chained header block '{0}' was found in store.", newBlockPair);
+                        onBlockDownloadedCallback(newBlockPair);
+                        continue;
+                    }
+                }
+
+                // If the list of blocks hashes requested for download are not consecutive we batch them
+                if (previousBlock != null && chainedHeaderBlock.ChainedHeader.Previous.HashBlock != previousBlock.ChainedHeader.HashBlock)
+                {
+                    this.logger.LogTrace("Block header '{0}' is not consecutive to previous block header '{1}'.", chainedHeaderBlock, previousBlock);
+                    this.DownloadBlocks(downloadRequests, this.ProcessDownloadedBlock);
+                    downloadRequests = new List<uint256>();
+                }
+
+                previousBlock = chainedHeaderBlock;
+                downloadRequests.Add(blockHash);
+                this.logger.LogTrace("Block hash '{0}' is queued for download.", blockHash);
+            }
+
+            this.DownloadBlocks(downloadRequests, this.ProcessDownloadedBlock);
+
+            this.logger.LogTrace("(-)");
+        }
+
+        /// <summary>
+        /// Processes items in <see cref="toDownloadQueue"/> and ask the block puller for blocks to download.
+        /// If the tree has too many unconsumed blocks or the download will wait, the amount of blocks to downloaded depends on the avg value in <see cref="IBlockPuller.AverageBlockSize"/>.
+        /// </summary>
+        /// <remarks>
+        /// Requests that have too many blocks will be split in batches.
+        /// </remarks>
+        private void ProcessDownloadQueueLocked()
+        {
+            this.logger.LogTrace("()");
+
+            const long MaxUnconsumedBlocksDataBytes = 1048576 * 10; // 10 mb
+
+            while (this.toDownloadQueue.Count > 0)
+            {
+                BlockDownloadRequest request = this.toDownloadQueue.Peek();
+
+                long freeMb = MaxUnconsumedBlocksDataBytes - this.chainedHeaderTree.UnconsumedBlocksDataBytes;
+                this.logger.LogTrace("There is {0} MB available do download is.", freeMb);
```

Comment:
It is fine to have the log in megabytes, but make sure you are comparing apples with apples.

Also write the sentence again, does not make sense.